w2/assignment2_2_2/Calculator/main.cpp

```diff
@@ -0,0 +1,283 @@
+/* Program can be used from command line or interface.
+ * Program uses external library for reading commandline arguments.
+ * Add, substract, multiply and divide are possible to use from command line.
+ * All are possible to use from interface and also encrypt and decrypt functions.
+ * Future development: 
+ *  1) Add error handling to commandline use
+ *  2) Add error handling for parsing string to double vector 
+ *  in case where seperator is something else than comma.
+ *  3) Combine user interface and command line reading if possible.
+*/
+
+#include "external/cxxopts/include/cxxopts.hpp"
+//#define NDEBUG
+#include <cassert>
+#include <iostream>
+
+#include "includes/utils.h"
+
+const std::string INVALID_CMD = "Invalid command.";
+const std::string INVALID_INPUT = "Invalid input.";
+const std::string INVALID_VALUES = "Invalid values.";
+const std::string INVALID_VALUES_AMOUNT = "Need at least two values";
+
+/* Function returns false if any of possible divisors equals to zero.
+ * Iteration starts from second value of vector, because first value (divinded)
+ * is allowed to equal zero.
+ */
+bool check_divider_not_zero(std::vector<double> vect)
+{
+    std::vector<double>::iterator it = vect.begin();
```

Comment:
tämän rivin voi ottaa pois. it muuttujan voi alustaa for loopissa sitten muotoa auto it = vect.begin()